### PR TITLE
Deprecate Pair and Triple

### PR DESCRIPTION
**Pull request by densh**
review by @retronym 
discussion https://groups.google.com/forum/#!topic/scala-internals/UCQBU-XPCy0

### CODE REVIEW

File: src/library/scala/Predef.scala [hunk 1]
```diff
@@ -26,7 +26,7 @@ import scala.io.ReadStdin
  *  [[scala.collection.immutable.Set]], and the [[scala.collection.immutable.List]]
  *  constructors ([[scala.collection.immutable.::]] and
  *  [[scala.collection.immutable.Nil]]).
- *  The types `Pair` (a [[scala.Tuple2]]) and `Triple` (a [[scala.Tuple3]]), with
+ *  Deprecated types `Pair` (a [[scala.Tuple2]]) and `Triple` (a [[scala.Tuple3]]), with
```
Comment (retronym) [inline src/library/scala/Predef.scala]: Let's remove this doc altogether.

File: src/library/scala/Predef.scala [hunk 2]
```diff
@@ -230,13 +230,17 @@ object Predef extends LowPriorityImplicits with DeprecatedPredef {
 
   // tupling ------------------------------------------------------------
 
+  @deprecated("Use built-in tuple syntax instead", "2.11.0")
```
Comment (retronym) [inline src/library/scala/Predef.scala]: Maybe: "Use built-in tuple syntax instead or Tuple2 instead?". That might help someone who is having trouble getting the parentheses correct, or who really doesn't want to use the syntactic sugar somewhere (I've known of people who actually prefer `def foo: Tuple2[A, B]` to `def foo: (A, B)`.)

Comment (retronym) [inline src/library/scala/Predef.scala]: The sugar also won't work in all cases, e.g: `class C[M[_, _]]; new C[Pair]`

Comment (densh) [inline src/library/scala/Predef.scala]: https://github.com/densh/scala/commit/7b9896f09f2d8452850059cf286900ce881d2f52#diff-532a778b9b259c18ed28fd43a22e44feR231

File: src/library/scala/Predef.scala (hunk 1 above)
Comment (densh) [inline src/library/scala/Predef.scala]: https://github.com/densh/scala/commit/7b9896f09f2d8452850059cf286900ce881d2f52#diff-532a778b9b259c18ed28fd43a22e44feL29

**Comment by soc** [comment]
```
- List[Pair[Name, Value]];
+ List[Tuple2[Name, Value]];
```

Why?!

**Comment by densh** [comment]
> Why?!

There are too many ways to do the same thing.

**Comment by adriaanm** [comment]
Agreed -- thanks for doing the dirty work, @densh!

**Comment by soc** [comment]
@densh Sorry, I meant why not `List[(Name, Value)]` instead?

**Comment by densh** [comment]
> why not List[(Name, Value)] instead?

I've been doing this refactoring semi-automatically. It's easy to rewrite `...Pair[...` into `...Tuple2[...` but harder if you need `...Pair[...]...` to become `...(...)...`.

**Comment by soc** [comment]
@densh Personally, I'd prefer `(...)` to having `Tuple2`/`Tuple3` in the source code. I have always considered the `TupleX`/`FunctionX`/`ProductX` stuff to be an implementation detail.

**Comment by soc** [comment]
For instance, if we ever decided to support the arity overloading of types like C#, we might get rid of the numbers forever.

**Comment by densh** [comment]
Wow, it finally builds as it should. /cc @retronym

**Comment by retronym** [comment]
LGTM

**Pull request merged**